pkg/server/apis/v1alpha1/types.go

```diff
@@ -45,3 +46,144 @@ type PersistentVolumeClaim struct {
+// IntegrationSource contains various external systems.
+// exactly one of its members must be set, and the member must equal with the integration's type.
+type IntegrationSource struct {
+	// SonarQube describes info about external system sonar qube, and is used for code scanning in CI.
+	SonarQube *SonarQubeSource `json:"sonarQube"`
+
+	// DockerRegistry describes info about external system docker registry, and is used to manager containers.
+	DockerRegistry *DockerRegistrySource `json:"dockerRegistry"`
+
+	// SCM describes info about external Source Code Management system, and is used to manager code.
+	SCM *SCMSource `json:"scm"`
+
+	// Cluster contains information about clusters.
+	// Users can define which cluster will be used to run workload,
+	// and clusters integrated here can be used to deploy application in CD tasks.
+	Cluster *ClusterSource `json:"cluster"`
+}
+
+// SonarQubeSource represents a code scanning tool for CI.
+type SonarQubeSource struct {
+	// Server represents the server address of sonar qube .
+	Server string `json:"server"`
+	// Token is the credential to access sonar server.
+	Token string `json:"token"`
+}
+
+// DockerRegistrySource represents a docker registry to manager containers.
+type DockerRegistrySource struct {
+	// Server represents the domain of docker registry.
+	Server string `json:"server"`
+	// User is a user of registry.
+	User string `json:"user"`
+	// Password is the password of the corresponding user.
+	Password string `json:"password"`
+}
+
+// SCMType defines the type of Source Code Management
+type SCMType string
+
+const (
+	// GitLab is the Gitlab scm
+	GitLab SCMType = "GitLab"
+	// GitHub is the GitHub scm
+	GitHub = "GitHub"
+	// SVN is the SVN scm
+	SVN = "SVN"
+)
+
+// SCMSource represents Source Code Management to manage code.
+type SCMSource struct {
+	// Type is the type of scm, e.g. GitLab, GitHub, SVN
+	Type SCMType `json:"type"`
+	// Server represents the domain of docker registry.
+	Server string `json:"server"`
+	// User is a user of the SCM.
+	User string `json:"user"`
+	// Password is the password of the corresponding user.
+	Password string `json:"password"`
+	// Token is the credential to access SCM.
+	Token string `json:"token"`
+}
+
+// ClusterSource contains info about clusters.
+type ClusterSource struct {
+	// Credential is the credential info of the cluster
+	Credential ClusterCredential `json:"credential"`
+	// InCluster describes whether the cluster is the control cluster itself
+	InCluster bool `json:"inCluster,omitempty"`
+	// Worker defines whether this cluster can be used to perform workload.
+	// True, will create namespace and pvc associated with tenant in the cluster.
+	// False, will delete namespace and pvc associated with tenant in the cluster.
+	Worker bool `json:"worker"`
+	// Namespace is the namespace where workload will run in. and if this is not nil, cyclone will
+	// use this namespace and not to create another one.
+	// Useful when 'Worker' is True.
```

Comment:
// Namespace is the namespace where workflows run in. If set, cyclone will use it directly, otherwise a new one will be created.
// It's used when 'Worker' is True.